### PR TITLE
Add docs around requests

### PR DESCRIPTION
**Pull request by justrudd**

### CODE REVIEW

**Review by josh-termly — APPROVED** [review]
from a technical perspective it looks good.  I am not great at English reviewing.

File: keys.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+Authentication for the Termly API uses 2 keys - a public and private. The public key is sent with each request. The private key is used to generate [signatures](signature.md) that Termly uses to verify the request.
+
+Each public key will have AT MOST one unexpired private key. If your key is compromised, you can immediately expire it and get a new key. If you are doing regularly scheduled key rolling, you can create a new private key for the public key. The old private key will be given an expiration 30 days from the date the new key is created. This expiration can be changed to be shorter or longer. Thirty days is just the default.
```
Comment (ciroque) [inline keys.md]: Confused, the tables in the internal documentation show that multiple keys can be active simultaneously...

Comment (ciroque) [reply keys.md]: "Unexpired" means there is an expiration date, not necessarily that it is expired...?

File: signature.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
+The Termly API requires the requestor to create a signature composing of pieces of the request. This signature ensures that request isn't changed from a `GET` to a `DELETE` if traveling through intermediaries. This signature also ensures that if a full request is logged and those logs are compromised at some later date, the request cannot be simply sent again and cause destructive behavior.
```
Comment (ciroque) [inline signature.md]: ...composed of...

Or 

...by composing pieces of...

File: signature.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+9ee59fbea7d22409648305e87b61e6d4257163017ffd19cf5c39007fdee1006f
+```
+
+The `9ee59...` string is the hex encoded SHA256 hash of the following JSON:
+
+```json
+[{"account_id":"acct_1234","email":"collaborator@example.com","role":"admin"}]
+```
+
+Please note that while the following are semantically the same, the would generate very different hashes.
```
Comment (ciroque) [inline signature.md]: ...they would generate...

**Review by ciroque — CHANGES_REQUESTED** [review]

File: keys.md (hunk 1 above)
Comment (justrudd) [reply keys.md]: In my head I was thinking unexpired meant missing.

**Review by ghost — APPROVED** [review]
Ship It!

**Pull request merged**